lang/py/src/avro/datafile.py

```diff
@@ -167,22 +175,26 @@ def _write_block(self):
 
       # write block contents
       uncompressed_data = self.buffer_writer.getvalue()
-      if self.get_meta(CODEC_KEY) == 'null':
+      codec = self.get_meta(CODEC_KEY)
+      if codec == NULL_CODEC:
```

Comment:
Maybe we should consider breaking up this code by some kind of standard interface?

```python
class Compressor(): # Abstract
    def compress(self, data):
        self._uncompressed = data
        self._compressed = self._compress(data)
        return self._compressed

    def __len__(self):
        return len(self._compressed)

    @abc.abstractmethod
    def _compress(self, data):
        ...
```